main/routines.c

```diff
@@ -202,11 +206,45 @@ extern const char *getExecutablePath (void)
 	return ExecutableProgram;
 }
 
+#ifdef HAVE_JANSSON
+void errorJSON (
+		const errorSelection selection, const char *const format, va_list ap)
+{
+	char *reason;
+	vasprintf (&reason, format, ap);
+
+	json_t *response = json_object ();
+	json_object_set_new (response, "error", json_string (reason));
+	if (selected (selection, WARNING))
+		json_object_set_new (response, "warning", json_true ());
+	if (selected (selection, FATAL))
+		json_object_set_new (response, "fatal", json_true ());
+	if (selected (selection, PERROR)) {
+		json_object_set_new (response, "errno", json_integer (errno));
+		json_object_set_new (response, "perror", json_string (strerror (errno)));
+	}
+	json_dumpf (response, stdout, 0);
+	fprintf (stdout, "\n");
+
+	json_decref (response);
+	free (reason);
+}
+#endif
+
 extern void error (
 		const errorSelection selection, const char *const format, ...)
 {
 	va_list ap;
 
+#ifdef HAVE_JANSSON
```

Comment:
(masatake should do): main part should provide pluggin interface for replacing error handler.